src/test/javascript/spec/component/course/course-group.component.spec.ts

```diff
@@ -75,130 +68,145 @@ describe('Course Management Detail Component', () => {
                 MockProvider(CourseManagementService),
                 MockProvider(UserService),
             ],
-        }).compileComponents();
-        fixture = TestBed.createComponent(CourseGroupComponent);
-        comp = fixture.componentInstance;
-        courseService = fixture.debugElement.injector.get(CourseManagementService);
-        userService = fixture.debugElement.injector.get(UserService);
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseGroupComponent);
+                comp = fixture.componentInstance;
+                courseService = TestBed.inject(CourseManagementService);
+                userService = TestBed.inject(UserService);
+            });
     });
 
-    afterEach(function () {
-        sinon.restore();
+    afterEach(() => {
+        jest.restoreAllMocks();
     });
 
     it('should initialize', () => {
         fixture.detectChanges();
-        expect(CourseGroupComponent).to.be.ok;
+        expect(CourseGroupComponent).toBeTruthy();
```

Comment:
Don't use `.toBeTruthy()`
Please refer to the coding guidelines for alternatives

Reply:
Also, what is the test actually? It looks to me that it checks whether the Type is truthy. Shouldn't it check this for the component itself?

Reply:
```suggestion
        expect(comp).toBeDefined();
```

Reply:
this would make a bit more sense (I saw something similar in other places), but I still question that this could actually fail

Reply:
Yes, probably was the intention. However, we should not use `.toBeDefined()` (see guidelines). This can create false negatives in some distinct cases.
We should test for specific attributes of that component in such cases rather than the sole existence.

Reply:
I used `toBeTruthy()` since it is used in some of the other refactored tests that already got merged.
What do you think about using `expect(CourseGroupComponent).not.toBe(null);`? This was suggested by @MDK5 in [4130](https://github.com/ls1intum/Artemis/pull/4130/files) and it was used in [4178](https://github.com/ls1intum/Artemis/pull/4178/files) instead of `.toBeDefined()`.

Reply:
Looking at it I think it actually makes no difference. The line is pointless anyways. TestBed.createComponent will never return null or undefined, and ngOnInit cannot change its subject. comp will always be defined/not-null/ok. One could consider to check some properties that are set after the initialization, which is essentially writing a completely new test. Personally I would just remove all tests like this, since they don't test anything. If there is nothing to check after a method call, we might as well don't run it.

Reply:
Alright. Thanks for your comment. Sure, I could remove the this test. 
What do you think @krusche and @julian-christl?

Reply:
Makes sense. As Martin says, this doesn't add any value.